src/workerd/api/sockets.h

```diff
@@ -69,27 +72,48 @@ struct SocketOptions {
 struct InitData {
   jsg::Ref<ReadableStream> readable;
   jsg::Ref<WritableStream> writable;
-  IoOwn<kj::PromiseFulfillerPair<void>> closeFulfiller;
+  IoOwn<kj::PromiseFulfillerPair<bool>> closeFulfiller;
+  IoOwn<kj::PromiseFulfillerPair<void>> jsCloseFulfiller;
```

Comment:
It looks like the caller consumes the promise half of this pair in order to create `jsCloseFulfillerFork`. Should this be `IoOwn<kj::PromiseFulfiller<void>>` to show that the promise is already consumed?